plugins/catalog-backend/src/permissions/rules/isEntityKind.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity } from '@backstage/catalog-model';
+import { EntitiesSearchFilter } from '../../catalog/types';
+
+/**
+ * A {@link CatalogPermissionRule} which filters for entities with a specified
+ * kind.
+ * @public
+ */
+export const isEntityKind = {
+  name: 'IS_ENTITY_KIND',
+  description: 'Allow entities with the specified kind',
+  apply(resource: Entity, kinds: string[]) {
+    return kinds
+      .map(kind => kind.toLocaleLowerCase('en-US'))
+      .some(kind => kind === resource.kind.toLocaleLowerCase('en-US'));
```

Comment:
Kinda feels like if we bother optimizing one of them we should optimize the other too xP

Reply:
Just so I make sure I understood this correctly, do you mean applying `.toLocaleLowerCase` to the other two rules as well?

Reply:
Was referring to the `.map` followed by the `.some` rather than dropping the `.map` and handling all case transformations in `.some`. We might as well take the optimization all the way

```ts
const resourceKind = resource.kind.toLocaleLowerCase('en-US');
return kinds
  .map(kind => kind.toLocaleLowerCase('en-US'))
  .some(kind => kind === resourceKind);
```

Reply:
Although I didn't realize we weren't doing anything nested here, so it wasn't really an optimization, just a way of writing it, so nvm :grin:

So just `kinds.some(kind => kind.toLocaleLowerCase('en-US') === resource.kind.toLocaleLowerCase('en-US'))`

Reply:
I see what you're saying and I do think it's better to revise as suggested. Done!